spec/classes/pulpcore_spec.rb

```diff
@@ -20,6 +20,24 @@
         end
       end
 
+      context 'with custom worker count' do
+        let :params do
+          {
+            worker_count: 5
+          }
+        end
+
+        it do
+          is_expected.to compile.with_all_deps
+          is_expected.to contain_service('pulpcore-worker@5')
+            .with_ensure('running')
+            .with_enable('true')
+          is_expected.to contain_service('pulpcore-worker@6')
```

Comment:
For this to be true, you need to set the fact:
```ruby
let(:facts) { super().merge(pulpcore_workers: ['pulpcore-worker@6.service']) }
```

I'd also make this a separate it block since it's something different you want to ensure from the previous one. The test with worker 5 ensures worker_count sets the right number of running workers. Worker 6 tests that the purging happens.

If you want to test more edge cases, you'd test with:
* worker count 5
  * fact missing
  * fact present, but empty
  * fact with 6 workers
  * fact with 6 workers

Reply:
@ekohl how could the fact be missing? my understanding is that the fact should always be present, and only if the current worker count is zero should it be empty.

Reply:
The choice I made in puppet-foreman was not to generate a fact if the directory didn't exist. Similar, you could check for the existence of the `pulp-worker@.service` file. One benefit of not presenting it is that you don't generate the fact for every host in your Puppet environment. That helps with fact storage, though I don't know how much it really matters.

Reply:
Thanks for the explanation @ekohl . In that case I think it does matter, c.f. https://access.redhat.com/solutions/3764171

I'll check for `pulpcore-worker@service` in the fact definition